src/code/z_bg_item.c

```diff
@@ -1,99 +1,144 @@
 }
 
-void BgCheck3_LoadMesh(GlobalContext* globalCtx, DynaPolyActor* actor, CollisionHeader* meshHeader) {
-    CollisionHeader* header;
+void DynaPoly_LoadMesh(GlobalContext* globalCtx, DynaPolyActor* dynaActor, CollisionHeader* meshHeader) {
+    CollisionHeader* header = NULL;
 
     header = NULL;
     BgCheck_RelocateMeshHeader(meshHeader, &header);
-    actor->bgId = BgCheck_AddActorMesh(globalCtx, &globalCtx->colCtx.dyna, actor, header);
+    dynaActor->bgId = BgCheck_AddActorMesh(globalCtx, &globalCtx->colCtx.dyna, dynaActor, header);
 }
 
-void BgCheck3_ResetFlags(DynaPolyActor* actor) {
-    actor->unk_158 = 0;
+void DynaPoly_ResetState(DynaPolyActor* dynaActor) {
+    dynaActor->stateFlags = DYNAPOLY_STATE_NONE;
 }
 
-void func_800CAE88(DynaPolyActor* actor) {
-    actor->unk_158 |= 1;
+void DynaPoly_SetRidingFallingState(DynaPolyActor* dynaActor) {
+    dynaActor->stateFlags |= DYNAPOLY_STATE_RIDING_FALLING;
 }
 
-void func_800CAE9C(DynaPolyActor* actor) {
-    actor->unk_158 |= 2;
+void DynaPoly_SetRidingMovingState(DynaPolyActor* dynaActor) {
+    dynaActor->stateFlags |= DYNAPOLY_STATE_RIDING_MOVING;
 }
 
-void func_800CAEB0(CollisionContext* colCtx, s32 index) {
-    DynaPolyActor* actor;
+void DynaPoly_SetRidingMovingStateByIndex(CollisionContext* colCtx, s32 index) {
+    DynaPolyActor* dynaActor;
 
-    actor = BgCheck_GetActorOfMesh(colCtx, index);
-    if (actor != (DynaPolyActor*)0x0) {
-        func_800CAE9C(actor);
+    dynaActor = BgCheck_GetActorOfMesh(colCtx, index);
+    if (dynaActor != NULL) {
+        DynaPoly_SetRidingMovingState(dynaActor);
     }
 }
 
-void func_800CAEE0(DynaPolyActor* actor) {
-    actor->unk_158 |= 4;
+void DynaPoly_SetRidingRotatingState(DynaPolyActor* dynaActor) {
+    dynaActor->stateFlags |= DYNAPOLY_STATE_RIDING_ROTATING;
 }
 
-void func_800CAEF4(CollisionContext* colCtx, s32 index) {
-    DynaPolyActor* actor;
+void DynaPoly_SetRidingRotatingStateByIndex(CollisionContext* colCtx, s32 index) {
+    DynaPolyActor* dynaActor;
 
-    actor = BgCheck_GetActorOfMesh(colCtx, index);
-    if (actor != (DynaPolyActor*)0x0) {
-        func_800CAEE0(actor);
+    dynaActor = BgCheck_GetActorOfMesh(colCtx, index);
+    if (dynaActor != NULL) {
+        DynaPoly_SetRidingRotatingState(dynaActor);
     }
 }
 
-void func_800CAF24(DynaPolyActor* actor) {
-    actor->unk_158 |= 8;
+void DynaPoly_SetSwitchPressedState(DynaPolyActor* dynaActor) {
+    dynaActor->stateFlags |= DYNAPOLY_STATE_SWITCH_PRESSED;
 }
 
-void func_800CAF38(DynaPolyActor* actor) {
-    actor->unk_158 |= 0x10;
+void DynaPoly_SetHeavySwitchPressedState(DynaPolyActor* dynaActor) {
+    dynaActor->stateFlags |= DYNAPOLY_STATE_HEAVY_SWITCH_PRESSED;
 }
 
-s32 func_800CAF4C(DynaPolyActor* actor) {
-    if (actor->unk_158 & 1) {
+s32 DynaPoly_IsInRidingFallingState(DynaPolyActor* dynaActor) {
+    if (dynaActor->stateFlags & DYNAPOLY_STATE_RIDING_FALLING) {
         return 1;
     } else {
         return 0;
     }
```

Comment:
You can replace those 1s and 0s to `true` and `false` now we know those functions are supposed to be boolean checks
```suggestion
        return true;
    } else {
        return false;
    }
```